### PR TITLE
azure: use Fedora 34 as a baseline

### PR DESCRIPTION
**Pull request by abbra**

### CODE REVIEW

**Comment by abbra** [comment]
I pulled @tiran's fixes for pylint and hostnames from https://github.com/freeipa/freeipa/pull/5504.

File: ipalib/util.py [hunk 1]
```diff
@@ -109,7 +109,10 @@ def json_serialize(obj):
 
 def verify_host_resolvable(fqdn):
     try:
-        if not resolve_ip_addresses(fqdn):
+        addresses = resolve_ip_addresses(fqdn)
+        if addresses:
+            return addresses
```
Comment (rcritten) [inline ipalib/util.py:114]: The return value doesn't seem to be used anywhere. This previously implicitly returned None or raised an exception.

**Comment by abbra** [comment]
Ok, this is the same issue as in March, we have some problem with d-bus in the container so that systemd-hostnamed is not able to respond:
```
021-05-04T12:12:02.0208240Z 1	[2021-05-04 12:12:01] INFO: 1_master_1: run: ['hostnamectl', 'set-hostname', 'master1.ipa.test']
2021-05-04T12:12:38.0820106Z 1	[2021-05-04 12:12:34] INFO: 1_master_1: result: 1
2021-05-04T12:12:38.0823203Z 1	[2021-05-04 12:12:34] ERROR: stderr: Could not set pretty hostname: Connection timed out
2021-05-04T12:12:38.0824328Z 1	[2021-05-04 12:12:34] 
2021-05-04T12:12:38.0825286Z 1	[2021-05-04 12:12:34] Traceback (most recent call last):
2021-05-04T12:12:38.0826822Z 1	[2021-05-04 12:12:34]   File "setup_containers.py", line 373, in <module>
2021-05-04T12:12:38.0828130Z 1	[2021-05-04 12:12:34]     controller.setup_hostname()
2021-05-04T12:12:38.0829355Z 1	[2021-05-04 12:12:34]   File "setup_containers.py", line 329, in setup_hostname
2021-05-04T12:12:38.0830171Z 1	[2021-05-04 12:12:34]     containers_group.setup_hostname()
2021-05-04T12:12:38.0831005Z 1	[2021-05-04 12:12:34]   File "setup_containers.py", line 186, in setup_hostname
2021-05-04T12:12:38.0831784Z 1	[2021-05-04 12:12:34]     cont.execute(cmd)
2021-05-04T12:12:38.0832731Z 1	[2021-05-04 12:12:34]   File "setup_containers.py", line 77, in execute
2021-05-04T12:12:38.0833548Z 1	[2021-05-04 12:12:34]     raise subprocess.CalledProcessError(
2021-05-04T12:12:38.0834515Z 1	[2021-05-04 12:12:34] subprocess.CalledProcessError: Command '['hostnamectl', 'set-hostname', 'master1.ipa.test']' returned non-zero exit status 1.
```

**Comment by abbra** [comment]
What's interesting is that the BASE_XMLRPC test suite actually worked past that part in its container:
https://dev.azure.com/freeipa/dd9bf4ec-d636-4558-8f6e-ee17af87f33b/_apis/build/builds/8775/logs/179
```

2021-05-04T12:11:54.5857746Z 1	[2021-05-04 12:11:54] INFO: 1_master_1: run: ['/bin/umount', '/etc/hosts']
2021-05-04T12:11:55.5538642Z 1	[2021-05-04 12:11:54] INFO: 1_master_1: result: 0
2021-05-04T12:11:55.5540884Z 1	[2021-05-04 12:11:54] INFO: 1_master_1: run: ['/bin/chmod', 'a-x', '/etc/hosts']
2021-05-04T12:11:55.5542190Z 1	[2021-05-04 12:11:55] INFO: 1_master_1: result: 0
2021-05-04T12:11:55.5543798Z 1	[2021-05-04 12:11:55] INFO: 1_master_1: run: ['/bin/bash', '-c', "echo -e '127.0.0.1 localhost\n::1 localhost\n172.19.0.2 master1.ipa.test\n2001:db8:1:1::2 master1.ipa.test' > /etc/hosts"]
2021-05-04T12:11:56.6177443Z 1	[2021-05-04 12:11:55] INFO: 1_master_1: result: 0
2021-05-04T12:11:56.6179088Z 1	[2021-05-04 12:11:55] INFO: 1_master_1: run: ['/bin/bash', '-c', "echo -e '' >> /etc/hosts"]
2021-05-04T12:11:56.6180283Z 1	[2021-05-04 12:11:56] INFO: 1_master_1: result: 0
2021-05-04T12:11:56.6181425Z 1	[2021-05-04 12:11:56] INFO: 1_master_1: run: ['/bin/umount', '/etc/hostname']
2021-05-04T12:11:57.6802444Z 1	[2021-05-04 12:11:56] INFO: 1_master_1: result: 0
2021-05-04T12:11:57.6807563Z 1	[2021-05-04 12:11:56] INFO: 1_master_1: run: ['/bin/chmod', 'a-x', '/etc/hostname']
2021-05-04T12:11:57.6808942Z 1	[2021-05-04 12:11:57] INFO: 1_master_1: result: 0
2021-05-04T12:11:57.6815438Z 1	[2021-05-04 12:11:57] INFO: 1_master_1: run: ['/bin/bash', '-c', "echo -e 'master1.ipa.test' > /etc/hostname"]
2021-05-04T12:11:58.7459576Z 1	[2021-05-04 12:11:57] INFO: 1_master_1: result: 0
2021-05-04T12:11:58.7461574Z 1	[2021-05-04 12:11:57] INFO: 1_master_1: run: ['hostnamectl', 'set-hostname', 'master1.ipa.test']
2021-05-04T12:12:17.1221777Z 1	[2021-05-04 12:12:16] INFO: 1_master_1: result: 0
2021-05-04T12:12:17.1223312Z 1	[2021-05-04 12:12:16] INFO: 1_master_1: run: ['/bin/umount', '/etc/resolv.conf']
2021-05-04T12:12:17.8982517Z 1	[2021-05-04 12:12:16] INFO: 1_master_1: result: 0
2021-05-04T12:12:17.8984818Z 1	[2021-05-04 12:12:16] INFO: 1_master_1: run: ['/bin/chmod', 'a-x', '/etc/resolv.conf']
2021-05-04T12:12:17.8985719Z 1	[2021-05-04 12:12:17] INFO: 1_master_1: result: 0
2021-05-04T12:12:17.8986919Z 1	[2021-05-04 12:12:17] INFO: 1_master_1: run: ['/bin/bash', '-c', "echo -e 'nameserver 8.8.8.8' > /etc/resolv.conf"]
2021-05-04T12:12:17.8987718Z 1	[2021-05-04 12:12:17] INFO: 1_master_1: result: 0
2021-05-04T12:12:17.8988968Z 1	[2021-05-04 12:12:17] INFO: 1_master_1: run: ['/bin/bash', '-c', "mkdir -p /etc/systemd/system/nis-domainname.service.d;echo '[Unit]' > /etc/systemd/system/nis-domainname.service.d/ipa-override.conf;echo 'ConditionVirtualization=!container' >> /etc/systemd/system/nis-domainname.service.d/ipa-override.conf"]
2021-05-04T12:12:17.8990324Z 1	[2021-05-04 12:12:17] INFO: 1_master_1: result: 0
2021-05-04T12:12:17.8991070Z 1	[2021-05-04 12:12:17] INFO: 1_master_1: run: ['systemctl', 'daemon-reload']
2021-05-04T12:12:18.9623567Z 1	[2021-05-04 12:12:17] INFO: 1_master_1: result: 0
```

**Comment by stanislavlevin** [comment]
Regarding hostnamectl issue.
```console
[root@docker ~]# journalctl -b | grep SECCOMP | grep -i dbus
мая 05 16:16:32 docker.test audit[4767]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=4767 comm="dbus-broker-lau" exe="/usr/bin/dbus-broker-launch" sig=0 arch=c000003e syscall=434 compat=0 ip=0x7f071664c15d code=0x7ffc0000
мая 05 16:16:32 docker.test audit[4800]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=4800 comm="dbus-broker-lau" exe="/usr/bin/dbus-broker-launch" sig=0 arch=c000003e syscall=434 compat=0 ip=0x7f5523b5b15d code=0x7ffc0000
```

https://man7.org/linux/man-pages/man2/pidfd_open.2.html:
> VERSIONS         top
       pidfd_open() first appeared in Linux 5.3.

Azure's kernel is 5.4.

This helps:
```diff
diff --git a/ipatests/azure/Dockerfiles/seccomp.json b/ipatests/azure/Dockerfiles/seccomp.json
index 350f1b4fa..599b98967 100644
--- a/ipatests/azure/Dockerfiles/seccomp.json
+++ b/ipatests/azure/Dockerfiles/seccomp.json
@@ -219,6 +219,7 @@
                                "open",
                                "openat",
                                "pause",
+                               "pidfd_open",
                                "pipe",
                                "pipe2",
                                "poll",
```

**Comment by abbra** [comment]
wonderful finding, thank you very much! Being heads down in a different issue (conference...) doesn't help with a clear sight ;)

I pushed an additional commit.

**Comment by stanislavlevin** [comment]
Next issue will be https://github.com/freeipa/freeipa/pull/5258#issuecomment-728908697 :)
I can take a look.

**Comment by stanislavlevin** [comment]
btw, libseccomp2 2.5.1 is in focal-updates (https://packages.ubuntu.com/focal-updates/libseccomp2).

**Comment by abbra** [comment]
@stanislavlevin thanks. If you find some more changes needed, you can push to this PR. I am going to look at more details myself after SambaXP conference ends (next week most likely).

**Comment by stanislavlevin** [comment]
I'm debugging in my pipelines to not waste shared CI resources:
https://github.com/stanislavlevin/freeipa/commits/debug_f34
https://dev.azure.com/slev0538/slev_/_build/results?buildId=119&view=results

**Comment by abbra** [comment]
Closing in favor of already merged PR #5790

**Pull request closed without merging**